src/components/ViewDeck/ViewDeck.js

```diff
@@ -1,94 +1,99 @@
-      payload: this.props.history.location.pathname.split("/")[2],
+      payload: history.location.pathname.split("/")[2],
     });
-  }
+  }, []);
 
-  editDeck = (deck) => {
-    console.log("in edit deck with:", deck);
-    const id = this.props.history.location.pathname.split("/")[2];
-    this.props.history.push(`/editdeck/${id}`);
+  //MANA CALCULATIONS
+  useEffect(() => {
+    const includedCards = cardList.filter(
+      (card) => card.deckid === selectedDeck.id
+    );
+    getManaCalculations(includedCards);
+  }, [cardList]);
+
+  const [includedCards, setIncludedCards] = useState([]);
+  const [cmc, setCmc] = useState([]);
+  const [averageCmc, setAverageCmc] = useState(0);
+  const [quantity, setQuantity] = useState(0);
+  const [featuredUri, setFeaturedUri] = useState("");
+  const [hoverCard, setHoverCard] = useState(
+    "https://i.stack.imgur.com/Vkq2a.png"
+  );
+  const [devotion, setDevotion] = useState({
+    White: 0,
+    Blue: 0,
+    Black: 0,
+    Red: 0,
+    Green: 0,
+    Gray: 0,
+  });
+
+  const NavToEditDeck = (deck) => {
+    history.push(`/editdeck/${deck.id}`);
   };
 
-  cardDisplay = (card) => {
-    //console.log('hovering on', card.name, card, card.api_data);
+  const setDisplayCard = (card) => {
     var parsedData = JSON.parse(card.api_data);
-    //    console.log(parsedData.image_uris.normal);
-
     if (parsedData.image_uris === undefined) {
-      this.setState({
-        hoverCard: `https://s3.thingpic.com/images/Kz/uF3amfCnYFLBggjUNr1sPRKi.jpeg`,
-      });
+      setHoverCard(
+        `https://s3.thingpic.com/images/Kz/uF3amfCnYFLBggjUNr1sPRKi.jpeg`
+      );
     } else {
-      this.setState({
-        hoverCard: parsedData.image_uris.normal,
-      });
+      setHoverCard(parsedData.image_uris.normal);
     }
   };
-
-  removeDisplay = (card) => {
-    // console.log('left:', card.name, card, card.api_data);
-    this.setState({
-      hoverCard: "https://i.stack.imgur.com/Vkq2a.png",
-    });
+  const removeDisplayCard = (card) => {
+    setHoverCard("https://i.stack.imgur.com/Vkq2a.png");
   };
 
-  countQty = (cards) => {
-    if (cards) {
-      var qty = 0;
-      cards.forEach((card) => {
-        //for each instance of a card, qtyToAdd = card.quantity
-        var qtyToAdd = card.quantity;
-        //console.log('qtyToAdd', qtyToAdd);
-        let totalqty = qty + qtyToAdd;
-        qty = totalqty;
-        //var totalCards = qty
-        //console.log('totalCards', totalCards);
-      });
-
-      console.log("new total:", qty);
-
-      if (this.state.deckQty !== qty) {
-        this.setState({
-          deckQty: qty,
-        });
-      }
-    }
+  const getManaCalculations = (cards) => {
+    // setIncludedCards(cards);
+    const qtyToSet = cards.reduce((qty, card) => qty + card.quantity, 0);
+    setQuantity(qtyToSet);
+    const parsedCards = parseCards(cards);
+    setIncludedCards(parsedCards);
+    const devotion = getDevotion(parsedCards);
+    setDevotion(devotion);
```

Comment:
There are two really good opportunities here.
1) your methods (parseCards, getDevotion, getCmc) all take an argument. So they don't rely on anything out of scope to return the desired value. So you could move the methods out of the component or even out of this file into a util or helper file. That way they are re-declared on every render. If they are in a util or helper file they would also be accessible to any other component that wanted to import them.
2) Every time you run a setState function (setQuantity, setIncludedCards, setDevotion, setFeaturedUri, and setCmc) the component re-renders. You run all of those independently in this function. So you are guaranteeing 5 render cycles just running this method once. How about we combine all of these things into one state object so we only have to run one setState function?
example:
```
const [deckState, setDeckState] = useState({
  quantity: 0,
  includedCards: [],
  devotion: {
    white,
    blue,
    red,
    green,
    black,
  },
  cmc: [],
  featuredUri: '',
});

const getManaCalculations = (cards) => {
  const qtyToSet = stuff;
  const parsedCards = things;
  const devotion = [array]
  setDeckState({
    ...deckState,
    quantity: qtyToSet,
    includedCards: parsedCards,
    devotion,
  });
};